libs/designer/src/lib/ui/panel/nodeDetailsPanel/tabs/mockResultsTab/mockResultsTab.tsx

```diff
@@ -37,24 +43,83 @@ export const MockResultsTab = () => {
   const mocks = useMocksByOperation(nodeName);
   const mocksValidationErrors = useMocksValidationErrors();
 
+  const [errorMessage, setErrorMessage] = useState<string>(mocks.errorMessage || '');
+  const [errorCode, setErrorCode] = useState<string>(mocks.errorCode || '');
+
   const filteredOutputs: OutputInfo[] = getFilteredOutputs(rawOutputs.outputs, nodeType);
+
+  useEffect(() => {
+    setErrorMessage(mocks.errorMessage || '');
+    setErrorCode(mocks.errorCode || '');
+  }, [mocks]);
+
   const onMockUpdate = useCallback(
     (id: string, newState: ChangeState, type: string) => {
       const { value, viewModel } = newState;
       const propertiesToUpdate = { value };
       if (!isNullOrUndefined(viewModel) && viewModel.arrayType === ArrayType.COMPLEX) {
         propertiesToUpdate.value = viewModel.uncastedValue;
       }
-      dispatch(updateMock({ operationName: nodeName, outputs: propertiesToUpdate.value ?? [], outputId: id, completed: true, type: type }));
+
+      if (id === 'errorMessage') {
+        setErrorMessage(value as unknown as string);
+      } else if (id === 'errorCode') {
+        setErrorCode(value as unknown as string);
+      }
+
+      if (mocks.actionResult === ActionResults.FAILED) {
+        dispatch(
+          updateMockFailure({
+            operationName: nodeName,
+            outputs: id === 'errorMessage' || id === 'errorCode' ? [] : propertiesToUpdate.value ?? [],
+            outputId: id,
+            completed: true,
+            type: type,
+            errorMessage: id === 'errorMessage' ? (value as unknown as string) : errorMessage,
+            errorCode: id === 'errorCode' ? (value as unknown as string) : errorCode,
+          })
+        );
+      } else {
+        dispatch(
+          updateMockSuccess({
+            operationName: nodeName,
+            outputs: propertiesToUpdate.value ?? [],
+            outputId: id,
+            completed: true,
+            type: type,
+          })
+        );
+      }
     },
-    [nodeName, dispatch]
+    [nodeName, dispatch, errorMessage, errorCode, mocks.actionResult]
   );
 
   const onActionResultUpdate = useCallback(
     (newState: ActionResultUpdateEvent): void => {
-      dispatch(updateActionResult({ operationName: nodeName, actionResult: newState.actionResult, completed: true }));
+      if (newState.actionResult === ActionResults.FAILED) {
+        dispatch(
+          updateActionResultFailure({
+            operationName: nodeName,
+            actionResult: newState.actionResult,
+            completed: true,
+            errorMessage,
+            errorCode,
+          })
+        );
+      } else {
+        dispatch(
+          updateActionResultSuccess({
+            operationName: nodeName,
+            actionResult: newState.actionResult,
+            completed: true,
+          })
+        );
+        // Clear error message and code when switching to success
+        setErrorMessage('');
```

Comment:
I feel that you will not need these ones anymore if you remove the states. What you can do is that on the `updateActionResultSuccess` function always set the state.errorMessage and state.errorCode as empty string. 
The intention for this is to save everything under react redux state instead of having single component states